tests/components/climacell/test_config_flow.py

```diff
@@ -0,0 +1,169 @@
+    with patch(
+        "homeassistant.components.climacell.config_flow.ClimaCell.realtime",
+        side_effect=CantConnectException,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=_get_config_schema(hass, MIN_CONFIG)(MIN_CONFIG),
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_user_flow_invalid_api(hass: HomeAssistantType) -> None:
+    """Test user config flow when API key is invalid."""
+    with patch(
+        "homeassistant.components.climacell.config_flow.ClimaCell.realtime",
+        side_effect=InvalidAPIKeyException,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=_get_config_schema(hass, MIN_CONFIG)(MIN_CONFIG),
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["errors"] == {CONF_API_KEY: "invalid_api_key"}
+
+
+async def test_user_flow_rate_limited(hass: HomeAssistantType) -> None:
+    """Test user config flow when API key is rate limited."""
+    with patch(
+        "homeassistant.components.climacell.config_flow.ClimaCell.realtime",
+        side_effect=RateLimitedException,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=_get_config_schema(hass, MIN_CONFIG)(MIN_CONFIG),
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["errors"] == {CONF_API_KEY: "rate_limited"}
+
+
+async def test_user_flow_unknown_exception(hass: HomeAssistantType) -> None:
+    """Test user config flow when unknown error occurs."""
+    with patch(
+        "homeassistant.components.climacell.config_flow.ClimaCell.realtime",
+        side_effect=UnknownException,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=_get_config_schema(hass, MIN_CONFIG)(MIN_CONFIG),
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "unknown"}
+
+
+async def test_options_flow(hass: HomeAssistantType) -> None:
+    """Test options config flow for climacell."""
+    user_config = _get_config_schema(hass)(MIN_CONFIG)
+    user_config[CONF_FORECAST_TYPES] = [NOWCAST]
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=user_config,
+        source=SOURCE_USER,
+        unique_id=_get_unique_id(hass, user_config),
+    )
+    entry.add_to_hass(hass)
+
+    assert not entry.options
+    assert CONF_TIMESTEP not in entry.data
+
+    result = await hass.config_entries.options.async_init(entry.entry_id, data=None)
```

Comment:
Make sure to set up the integration before starting the options flow. Use `hass.config_entries.async_setup`.